backend/main/models/vote.go

```diff
@@ -297,6 +297,16 @@ func (v *Vote) ValidateChoice(proposal Proposal) error {
 	return nil
 }
 
+func CheckForMaxWeight(maxWeight float64, balance uint64) bool {
+	floatBalance := float64(balance)
+
+	if maxWeight != 0 && floatBalance >= maxWeight {
```

Comment:
oh i see, so you are saying if max_weight is 0, then skip the check

Reply:
yah, arrived at this to avoid max_weight ever being `nil`. if max_weight was not set to 0 somewhere it was becoming difficult to do checks against it. I could try something like
```
if p.Max_weight != nil {
// then enforce it here 
}
```

Reply:
updated with a refactor!